test/test_beanstalk_utils.py

```diff
@@ -352,3 +353,43 @@ def test_compute_prd_env_for_project():
             mock_describer.side_effect = _mocked_describe_beanstalk_environments
             assert bs._compute_prd_env_for_project('cgap') == 'cgap-env-2'
             assert bs._compute_prd_env_for_project('ff') == 'ff-env-2'
+
+
+@pytest.mark.parametrize('options, expected', [
+    ([
+         {
+            'Namespace': bs.ENV_VARIABLE_NAMESPACE,
+            'OptionName': 'super_secret',
+            'Value': 'i am secret'
+         }
+     ], {'super_secret': 'i am secret'}),
+    ([
+         {
+            'Namespace': bs.ENV_VARIABLE_NAMESPACE,
+            'OptionName': 'super_secret',
+            'Value': 'i am secret'
+         },
+         {
+            'Namespace': 'something else',
+            'OptionName': 'not_secret',
+            'Value': 'i dont care about this value'
+         }
+     ], {'super_secret': 'i am secret'}),
+    ([
+         {
+            'Namespace': 'identifier',
+            'OptionName': 'something',
+            'Value': 'important'
+         },
+         {
+            'Namespace': 'something else',
+            'OptionName': 'not_secret',
+            'Value': 'i dont care about this value'
+         }
+     ], {})
+])
```

Comment:
Should probably have one more case that tests that you're accumulating two values when both are in `bs.ENV_VARIABLE_NAMESPACE`, but this is fine either way.